engine/src/main/java/io/camunda/zeebe/engine/state/immutable/CompensationSubscriptionState.java

```diff
@@ -17,4 +18,10 @@ CompensationSubscription get(
 
   Set<CompensationSubscription> findSubscriptionsByProcessInstanceKey(
       String tenantId, long processInstanceKey);
+
+  Optional<CompensationSubscription> findSubscriptionByCompensationHandlerId(
+      String tenantId, long processInstanceKey, String compensationHandlerId);
+
+  Set<CompensationSubscription> findSubscriptionByThrowEventInstanceKey(
```

Comment:
:wrench: The method returns more than one subscription.

```suggestion
  Set<CompensationSubscription> findSubscriptionsByThrowEventInstanceKey(
```